domain/relation/state/relation.go

```diff
@@ -166,6 +127,110 @@ func (st *State) AddRelation(ctx context.Context, epIdentifier1, epIdentifier2 r
 	})
 }
 
+// SetRelationWithID establishes a relation between two endpoints identified
+// by ep1 and ep2 and returns the relation UUID. Used for migration
+// import.
+func (st *State) SetRelationWithID(
+	ctx context.Context,
+	epIdentifier1, epIdentifier2 relation.CandidateEndpointIdentifier,
+	id uint64,
+) (corerelation.UUID, error) {
+	var relUUID corerelation.UUID
+	db, err := st.DB()
+	if err != nil {
+		return relUUID, errors.Capture(err)
+	}
+
+	return relUUID, db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		// Infers endpoint, ie get both application_endpoint_uuid.
+		ep1, ep2, err := st.inferEndpoints(ctx, tx, epIdentifier1, epIdentifier2)
```

Comment:
We shouldn't need to infer endpoints on relations we are importing, we know what format we are getting them in, we should be able to skip this step.

Reply:
Its a ton of extra code to call when you could just do this:
```
SELECT ae.uuid
FROM application_endpoint ae
JOIN application a ON a.uuid = ae.application_uuid
JOIN charm_relation cr ON ae.charm_relation_uuid = cr.uuid
WHERE cr.name = ?
WHERE a.name = ?
```

I really think we should stop calling infer endpoints here, its very misleading to read, it suggests we are importing incomplete data which is not the case.

Reply:
done